upload-documentation.sh

```diff
@@ -138,33 +115,65 @@ for (( i=1; i<=$MAX_ATTEMPTS; i++ ))
 do
```

Comment:
^ note I went with the whole thing being inside the loop and if the push fails, it should remove the cloned repo in that tmp dir and start from 0